api/types/okta.go

```diff
@@ -396,6 +396,79 @@ func (o *OktaAssignmentV1) CheckAndSetDefaults() error {
 	return nil
 }
 
+// IsEqual determines if two okta assignment resources are equivalent to one another.
+func (o *OktaAssignmentV1) IsEqual(i interface{}) bool {
+	ptr, ok := i.(*OktaAssignmentV1)
+	if ok {
+		if o == nil && ptr == nil {
+			return true
+		}
+		if o != nil && ptr != nil {
+			return o.isEqual(*ptr)
+		}
+		return false
+	}
+
+	other, ok := i.(OktaAssignmentV1)
+	if ok {
+		return o.isEqual(other)
+	}
+
+	return false
+
+}
+
+func (o OktaAssignmentV1) isEqual(other OktaAssignmentV1) bool {
```

Comment:
Do we need to make two copies of `OktaAssignmentV1` every time we do an equality check?

Reply:
Fair enough. I'll invert this relationship here.